### PR TITLE
feat(api,vcs): use new vcs project

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17369.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17370.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17375.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17377.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17380.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17383.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17387.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17388.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17388.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17389.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17390.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17391.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by richardlt — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17392.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17394.0 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ■   
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test SDK Rust ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17394.0 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ■   
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test SDK Rust ■   
  * Test UI ■

File: engine/vcs/bitbucketserver/http.go [hunk 1]
```diff
@@ -114,11 +111,17 @@ func (c *bitbucketClient) do(ctx context.Context, method, api, path string, para
...
-	if opts != nil && opts.asUser && c.token != "" {
+	var cacheKey string
+	if c.vcsProject != nil && c.vcsProject.Auth["token"] != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.vcsProject.Auth["token"]))
+		cacheKey = cache.Key("vcs", "bitbucket", "request", req.URL.String(), c.vcsProject.Auth["username"])
+	} else if opts != nil && opts.asUser && c.token != "" { // DEPRECATED VCS
 		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.token))
-	} else {
-		if err := c.consumer.Sign(req, token); err != nil {
+		cacheKey = cache.Key("vcs", "bitbucket", "request", req.URL.String(), c.token)
```
Comment (richardlt) [inline engine/vcs/bitbucketserver/http.go:120]: Create a hash for the token

**Review by richardlt — CHANGES_REQUESTED** [review]

**Review by richardlt — APPROVED** [review]

File: engine/api/application.go [hunk 2]
```diff
@@ -247,6 +243,10 @@ func (api *API) getApplicationVCSInfosHandler() service.Handler {
 		}
 		resp.Remotes = remotes
 
+		if err := tx.Commit(); err != nil {
```
Comment (fsamin) [inline engine/api/application.go:246]: Why did we need a DB transaction for this handler ?
It seems to be only just return data from the vcs

File: engine/api/workflow_event.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func (api *API) WorkflowSendEvent(ctx context.Context, proj sdk.Project, report
 		event.PublishWorkflowNodeRun(ctx, *nr, wr.Workflow, eventsNotif)
 		e := &workflow.VCSEventMessenger{}
 		if err := e.SendVCSEvent(ctx, db, api.Cache, proj, *wr, wnr); err != nil {
-			log.Warn(ctx, "WorkflowSendEvent> Cannot send vcs notification")
+			log.Warn(ctx, "WorkflowSendEvent> Cannot send vcs notification err:%v", err)
```
Comment (fsamin) [inline engine/api/workflow_event.go:64]: + ctx := log.ContextWithStrackTrace(ctx, err)

File: engine/vcs/bitbucketserver/bitbucketserver.go [hunk 4]
```diff
@@ -11,14 +11,16 @@ import (
 
 // bitbucketClient is a bitbucket wrapper for CDS vcs. interface
 type bitbucketClient struct {
-	consumer          bitbucketConsumer
-	accessToken       string
-	accessTokenSecret string
+	vcsProject        *sdk.VCSProject
```
Comment (fsamin) [inline engine/vcs/bitbucketserver/bitbucketserver.go:14]: the client doesn't have to hold this

**Review by fsamin — CHANGES_REQUESTED** [review]

File: engine/api/application.go (hunk 2 above)
Comment (yesnault) [reply engine/api/application.go:246]: `AuthorizedClient` takes a `gorpmapper.SqlExecutorWithTx` in his client. This client can be used to update the token / created date with `checkAccessToken`
All this code can be removed when the old vcs will be deleted.

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17403.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/vcs/bitbucketserver/bitbucketserver.go (hunk 4 above)
Comment (yesnault) [reply engine/vcs/bitbucketserver/bitbucketserver.go]: fixed

File: engine/api/workflow_event.go (hunk 3 above)
Comment (yesnault) [reply engine/api/workflow_event.go:69]: fixed

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17406.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✘

**Review by fsamin — APPROVED** [review]

**Pull request merged**